src/App.js

```diff
@@ -63,73 +63,187 @@ const useStyles = makeStyles({
+      isNonNegOperator.test(input)
+    ) {
+      console.log('condition 4');
+      setEquation(equation.slice(0, -2) + input);
+      setCurrentToken(input);
+    } else if (
+      isOperator.test(currentToken) &&
+      isOperator.test(equation.slice(-1))
+    ) {
+      console.log('condition 5 operator');
+      setCurrentToken(input);
+      setEquation(equation.slice(0, -1) + input);
+    } else if (equation === '') {
+      setCurrentToken(input);
+      setEquation(input);
+    } else if (equation.includes('=')) {
+      console.log('handleOperator condition 6');
+      setCurrentToken(input);
+      setEquation(answer + input);
     }
-    if (isClear) {
-      setCurrentVal('0');
-      setPrevVal(null);
-      setEquation([]);
-    } else if (currentVal === '0' && prevVal === null) {
-      setCurrentVal(input);
-      setPrevVal(input);
+  };
+
+  const handleOperand = ({ currentTarget: { value } }) => {
+    const input = value;
+    if (
+      (isOperand.test(currentToken) && equation === '') ||
+      (currentToken === '0' && equation === '0')
+    ) {
+      console.log('handleOperand condition 1');
+      setCurrentToken(input);
       setEquation(input);
-    } else {
-      setPrevVal(currentVal);
-      console.log('prevVal:', prevVal);
-      setCurrentVal(equation + input);
+    } else if (isOperand.test(currentToken) && currentToken !== '0') {
+      console.log('handleOperand condition 2');
+      setCurrentToken(currentToken + input);
+      setEquation(equation + input);
+    } else if (isOperator.test(currentToken)) {
+      console.log('handleOperand condition 3');
+      setCurrentToken(input);
       setEquation(equation + input);
     }
   };
 
+  const handleDecimal = ({ currentTarget: { value } }) => {
+    const decimal = value;
+    console.log('decimal:', value);
+    console.log('currentToken for decmial:', currentToken);
+    if (
+      !currentToken.toString().includes('.') &&
+      isOperand.test(currentToken) &&
+      currentToken !== '0'
+    ) {
+      console.log('decimal condition 1:');
+      setCurrentToken(currentToken.toString() + decimal);
+      setEquation(equation + decimal);
+    } else if (isOperator.test(currentToken)) {
+      console.log('test', isOperator.test('.'));
+      console.log('decimal condition 2');
+      console.log('currentToken', currentToken);
+      setCurrentToken(`0${decimal}`);
+      setEquation(`${equation}0${decimal}`);
+    }
+  };
+
+  const handleClear = () => {
+    setEquation('');
+    setPrevInput('');
+    setCurrentToken('0');
+  };
+
+  const handleEval = ({ currentTarget: { value } }) => {
+    const input = value;
+    console.log('eval:', input);
+    // const numEqualsNum = /\d+=\d+/;
+    const startsWithMultiOrDiv = /^[*/]/;
+    if (!equation.includes('=') && !startsWithMultiOrDiv.test(equation)) {
+      console.log('handleEval condition 1');
+      // eslint-disable-next-line no-eval
+      const result = eval(equation);
```

Comment:
I think this is fine for this project, but `eval` in any language that evaluates an arbitrary string of executable code is considered highly dangerous to use. You may have seen [Never use eval!](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/eval#never_use_eval!) before. If you want an extra challenge, it's a fun challenge to do the evaluation of the equation input string in place of using `eval`.